### ISSUE TITLE
`gccForLibs` can be updated for `llvmPackages_X.stdenv` for `i686Linux`

### ISSUE THREAD

**Issue by peperunas** [NixOS/nixpkgs#146865]
###### Checklist

- [X] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [X] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name: `i686Linux.llvmPackages_XX.stdenv`

Currently, the `stdenv` for `clang` based `stdenv`s is using an outdated `gccForLibs`. 
I have tested `gcc10` and `gcc11` with `llvmPackages_10` and `llvmPackages_11` and I did not encounter any issues when compiling an LLVM source tree.

This is what I have tested:

```nix
  stdenv_clang = pkgs.overrideCC pkgs.stdenv
    (pkgs.clang_10.override ({ gccForLibs = pkgs.gcc.cc; }));
```

current version: `gcc 6.5`
desired version: `gcc10`+

###### Notify maintainers

maintainers:

###### Note for maintainers

Please tag this issue in your PR.

**Comment by layus** [comment]
@Ericson2314, I think this one may interest you as you worked hard on refactoring the bootstrap and cross-compiling architecture. Would it make sense to remove https://github.com/NixOS/nixpkgs/blob/3766a8f7f41158ac37595466b8d58d4f3d7c39dc/pkgs/top-level/all-packages.nix#L13079-L13088 altogether ? or maybe keep only gcc6 for very old clang ?

**Comment by peperunas** [comment]
This PR might be related:

https://github.com/NixOS/nixpkgs/pull/163044

**Comment by peperunas** [comment]
Closed by #163044
